app/controllers/settings/import_controller.rb

```diff
@@ -8,23 +8,30 @@ def myanimelist
     hide_cover_image
 
     if params[:animelist]
+      file = params[:animelist]
       begin
-        if params[:animelist].content_type == "text/xml"
-          xml = params[:animelist].read
-        else
-          gz = Zlib::GzipReader.new(params[:animelist])
+        # Check the magic number for gzip because some browsers are stupid
+        # Many users are uploading zipped-up lists with a text/xml MIME
+        if file.readpartial(3) == 0x1F_8B_08
+          gz = Zlib::GzipReader.new(file)
           xml = gz.read
           gz.close
+        elsif file.content_type.include?('xml')
+          xml = file.read
+        else
+          raise "Unknown format"
         end
 
         xml = XMLCleaner.clean(xml)
 
+        raise "Blank file" if xml.blank?
+
         current_user.update_column :mal_import_in_progress, true
         MyAnimeListImportApplyWorker.perform_async(current_user.id, xml)
 
         mixpanel.track "Imported from MyAnimeList", {email: current_user.email} if Rails.env.production?
-      rescue
-        flash[:error] = "There was a problem importing your anime list. Please send an email to <a href='mailto:vikhyat@hummingbird.me'>vikhyat@hummingbird.me</a> with the file you are trying to import.".html_safe
+      rescue Exception => e
+        flash[:error] = "There was a problem importing your anime list (#{e.message}).  Please send an email to <a href='mailto:vikhyat@hummingbird.me'>vikhyat@hummingbird.me</a> with the file you are trying to import.".html_safe
```

Comment:
I agree that it's not ideal to show them the raw error message, but we need to expose the most basic error info to the user somehow and this was just a quick hacked-together PR for avoiding (and helping debug) common failures.

That being said, it's also not as if a user will ever see this anyways: very few pages display the flash messages anymore.  I'm working on a better solution right now.